### PR TITLE
Fix eslint rule `eqeqeq`

### PR DESCRIPTION
**Pull request by scinos**
#### Changes proposed in this Pull Request

* Fixes eslint rule `eqeqeq` across the whole repo
* While working on the affected files I used eslint auto-fixer to fix other minor problems.

#### Testing instructions

N/A

### CODE REVIEW

File: apps/notifications/src/panel/templates/block-user.jsx [hunk 1]
```diff
@@ -65,13 +64,12 @@ export class UserBlock extends React.Component {
 	};
 
 	render() {
-		var grav = this.props.block.media[ 0 ],
-			home_url = '',
-			home_title = '',
-			timeIndicator,
-			homeTemplate,
-			followLink,
-			noteActions;
```
Comment (scinos) [inline apps/notifications/src/panel/templates/block-user.jsx:74]: `noteActions` was unused.

File: apps/notifications/src/panel/templates/block-user.jsx [hunk 2]
```diff
@@ -110,8 +108,8 @@ export class UserBlock extends React.Component {
 		}
 
 		if ( home_title ) {
-			var homeClassName =
-				timeIndicator != '' ? 'wpnc__user__meta wpnc__user__bulleted' : 'wpnc__user__meta';
+			const homeClassName =
+				timeIndicator !== '' ? 'wpnc__user__meta wpnc__user__bulleted' : 'wpnc__user__meta';
```
Comment (scinos) [inline apps/notifications/src/panel/templates/block-user.jsx:112]: `timeIndicator` is defined a few lines above and it is either a React component or an string

File: apps/notifications/src/panel/templates/body.jsx [hunk 3]
```diff
@@ -52,16 +52,16 @@ export const NoteBody = createReactClass( {
...
+		let hasReplyBlock;
 
 		if ( note.meta && note.meta.ids.reply_comment ) {
 			hasReplyBlock =
 				note.body.filter( function ( block ) {
 					return (
 						block.ranges &&
 						block.ranges.length > 1 &&
-						block.ranges[ 1 ].id == note.meta.ids.reply_comment
+						block.ranges[ 1 ].id === note.meta.ids.reply_comment
```
Comment (scinos) [inline apps/notifications/src/panel/templates/body.jsx:64]: I couldn't find docs about the type of `reply_comment`, but https://github.com/Automattic/wp-calypso/blob/fix/lint-app-notifications/apps/notifications/src/panel/templates/functions.jsx#L215 is doing the same check and they use `===` there, so I guess it is safe.

File: apps/notifications/src/panel/templates/comment-reply-input.jsx [hunk 4]
```diff
@@ -17,11 +17,8 @@ import { wpcom } from '../rest-client/wpcom';
 import { bumpStat } from '../rest-client/bump-stat';
 import { formatString, validURL } from './functions';
 
-var debug = require( 'debug' )( 'notifications:reply' );
-var { recordTracksEvent } = require( '../helpers/stats' );
-
-const hasTouch = () =>
```
Comment (scinos) [inline apps/notifications/src/panel/templates/comment-reply-input.jsx:23]: Unused

File: apps/notifications/src/panel/templates/comment-reply-input.jsx [hunk 5]
```diff
@@ -150,26 +147,26 @@ const CommentReplyInput = createReactClass( {
...
+		let statusMessage;
+		const successMessage = this.props.translate( 'Reply posted!' );
+		const linkMessage = this.props.translate( 'View your comment.' );
 
 		if ( event ) {
 			event.preventDefault();
 		}
 
-		if ( '' == this.state.value ) return;
+		if ( '' === this.state.value ) return;
```
Comment (scinos) [inline apps/notifications/src/panel/templates/comment-reply-input.jsx:161]: `this.state.value` is set to an empty string as default

File: apps/notifications/src/panel/templates/block-user.jsx [hunk 6]
```diff
@@ -147,28 +145,27 @@ export class UserBlock extends React.Component {
 		if ( home_url ) {
 			return (
 				<div className="wpnc__user">
-					<a className="wpnc__user__site" href={ home_url } target="_blank">
+					<a className="wpnc__user__site" href={ home_url } target="_blank" rel="noreferrer">
```
Comment (tyxla) [inline apps/notifications/src/panel/templates/block-user.jsx:148]: Noting that we generally prefer ` rel="noreferrer noopener"` instead of just `noreferrer`.

**Review by tyxla — COMMENTED** [review]
Changes look sensible to me. Just left a small hint.

Would be really nice to get some testing instructions here. 

I've done some smoke testing on the notifications panel in the meantime. Looks good!

File: apps/notifications/src/panel/templates/block-user.jsx (hunk 6 above)
Comment (scinos) [reply apps/notifications/src/panel/templates/block-user.jsx:148]: I'll bring it back to be consistent, but it is not required according to https://github.com/yannickcr/eslint-plugin-react/issues/2022. 

tldr: The eslint rule was checking for both, the spec says `noreferrer` is enough and someone did a test in many browsers and found that is the case.

Comment (tyxla) [reply apps/notifications/src/panel/templates/block-user.jsx:148]: Back when I was looking at why we're adding both `noreferrer` and `noopener` I recall I didn't find much information about `noopener`, nor browser support information. I guess it's safe to say it doesn't hurt to keep it in place just in case.

Comment (jsnajdr) [reply apps/notifications/src/panel/templates/block-user.jsx:148]: The https://github.com/yannickcr/eslint-plugin-react/issues/2022 discussion concludes that there are browsers that remove `window.opener` only on `rel="noreferrer"` and _won't_ remove it on `rel="noreferrer noopener"`, because they support `noreferrer` and don't understand the `noopener` attribute, so fail to parse the latter 🤦 

The spec (and [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/open#noreferrer)) says that `noreferrer` should also remove `window.opener`, and browsers either do it, or don't do it at all, even with `noopener` added.

There are no browsers that would:
- not remove `window.opener` on mere `rel="noreferrer"`
- *and* would start removing it after adding `rel="noopener"`

Therefore, `noopener` never has any useful effect and `rel="noreferrer"` is always enough.

Things were probably different as late as July 2019 last year, because Firefox didn't support `noreferrer` until v68 released that month. But today, the above describes the state of the art 🙂

Comment (tyxla) [reply apps/notifications/src/panel/templates/block-user.jsx:148]: Thanks for the context! What a mess! 😅

File: apps/notifications/src/panel/templates/comment-reply-input.jsx [hunk 7]
```diff
@@ -140,7 +137,7 @@ const CommentReplyInput = createReactClass( {
 
 		// Reset the field if there's no valid user input
 		// The regex strips whitespace
-		if ( '' == this.state.value.replace( /^\s+|\s+$/g, '' ) ) {
+		if ( '' === this.state.value.replace( /^\s+|\s+$/g, '' ) ) {
```
Comment (jsnajdr) [inline apps/notifications/src/panel/templates/comment-reply-input.jsx:140]: `value.trim()` should do exactly the same thing: remove whitespace (including tabs, newlines etc) from both ends.

File: apps/notifications/src/panel/templates/status-bar.jsx [hunk 8]
```diff
@@ -34,14 +34,14 @@ export class StatusBar extends React.Component {
...
 
-		var component = this;
+		const component = this;
 
 		/* We only want this to appear for a bit, then disappear */
-		var timeout = window.setTimeout(
+		const timeout = window.setTimeout(
 			function () {
 				component.disappear();
 			},
```
Comment (jsnajdr) [inline apps/notifications/src/panel/templates/status-bar.jsx:47]: Can be replaced with `() => { this.disappear(); }`, with the desired `this` binding by default.

File: apps/notifications/src/panel/templates/summary-in-list.jsx [hunk 9]
```diff
@@ -15,8 +15,7 @@ import ImagePreloader from './image-loader';
 
 import { html } from '../indices-to-html';
 
-var debug = require( 'debug' )( 'notifications:summary-in-list' );
-var { recordTracksEvent } = require( '../helpers/stats' );
+const { recordTracksEvent } = require( '../helpers/stats' );
```
Comment (jsnajdr) [inline apps/notifications/src/panel/templates/summary-in-list.jsx:18]: Could also be `import`. I guess some automated codemod missed it before, as it's not present in the initial `import` block.

File: apps/notifications/src/panel/templates/undo-list-item.jsx [hunk 10]
```diff
@@ -15,7 +15,7 @@ import { bumpStat } from '../rest-client/bump-stat';
 
 import Gridicon from './gridicons';
 
-var { recordTracksEvent } = require( '../helpers/stats' );
+const { recordTracksEvent } = require( '../helpers/stats' );
```
Comment (jsnajdr) [inline apps/notifications/src/panel/templates/undo-list-item.jsx:18]: `import`

File: apps/notifications/src/panel/templates/undo-list-item.jsx [hunk 11]
```diff
@@ -90,15 +90,15 @@ export class UndoListItem extends React.Component {
 	};
 
 	spamComment = () => {
-		var comment = wpcom()
+		const comment = wpcom()
 			.site( this.props.note.meta.ids.site )
 			.comment( this.props.note.meta.ids.comment );
-		var component = this;
+		const component = this;
```
Comment (jsnajdr) [inline apps/notifications/src/panel/templates/undo-list-item.jsx:96]: This looks redundant. There's not even a candidate for arrow function that would use the variable. Just one simple statement that could be converted to `this.finishExecution()`

**Review by jsnajdr — APPROVED** [review]
It's nice to see that the long-neglected notifications code finally gets some love 💟 I noticed several other opportunities for easy modernization. They are not blockers at all.

**Comment by scinos** [comment]
Stale and no plans to work on it in the near future
